src/index.ts

```diff
@@ -65,21 +92,21 @@ class Majo {
     })
   }
 
-  transform(relative, fn) {
+  transform(relative: string, fn: TransformFn) {
     const contents = this.files[relative].contents.toString()
     const result = fn(contents)
-    if (!result.then) {
+    if (typeof result === 'string') {
       this.files[relative].contents = Buffer.from(result)
       return
     }
-    return result.then(newContents => {
+    return result.then((newContents: string) => {
```

Comment:
I think it's not making sense, when user write a transform function that returns a number, the error message will be `result.then is not a function`. the user will be very confused so he must to read the source code.